### PR TITLE
ECDSA keep reward specialization

### PR DESCRIPTION
**Pull request by eth-r**
New implementation of ECDSA keep rewards based on the abstract rewards contract.

### CODE REVIEW

File: solidity/contracts/ECDSAKeepRewards.sol [hunk 1]
```diff
@@ -0,0 +1,96 @@
+pragma solidity ^0.5.4;
+
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "@keep-network/keep-core/contracts/Rewards.sol";
+
+contract ECDSAKeepRewards is Rewards {
+    uint256 constant MAX_UINT20 = 1 << 20 - 1;
```
Comment (benlongstaff) [inline solidity/contracts/ECDSAKeepRewards.sol:9]: where does this get used? I couldn't find it in keep-ecdsa or keep-core?

File: solidity/contracts/ECDSAKeepRewards.sol [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    uint256 constant _minimumKeepsPerInterval = 2;
```
Comment (benlongstaff) [inline solidity/contracts/ECDSAKeepRewards.sol:10]: should this number be higher? i'm totally ok with it staying at 2 though

File: solidity/contracts/ECDSAKeepRewards.sol [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    IBondedECDSAKeepFactory factory;
+
+    constructor (
+        address _token,
+        address factoryAddress,
+        uint256 _firstIntervalStart,
+        uint256[] memory _intervalWeights
```
Comment (benlongstaff) [inline solidity/contracts/ECDSAKeepRewards.sol:17]: why is intervalWeights as a constructor param and _minimumKeepsPerInterval a constant?

File: solidity/contracts/ECDSAKeepRewards.sol (hunk 1 above)
Comment (pdyraga) [reply solidity/contracts/ECDSAKeepRewards.sol:9]: Same here, it looks unused so I'll drop it.

File: solidity/contracts/ECDSAKeepRewards.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (pdyraga) [reply solidity/contracts/ECDSAKeepRewards.sol:10]: It's going to be higher, I am pretty sure this value is just a temporary one. I'll drop a TODO here.

File: solidity/contracts/ECDSAKeepRewards.sol (hunk 3 above)
Comment (pdyraga) [reply solidity/contracts/ECDSAKeepRewards.sol:17]: Good point. I think we are going to make all those parameters constants to minimize the risk of making a mistake during deployment.

**Comment by pdyraga** [comment]
Closing in favor of https://github.com/keep-network/keep-ecdsa/pull/567 and https://github.com/keep-network/keep-ecdsa/pull/571 that were based on this code.

**Pull request closed without merging**